src/main/java/com/spotify/autoscaler/api/ClusterResources.java

```diff
@@ -211,28 +211,32 @@ public Response deleteCluster(
   }
 
   @PUT
-  @Path("load")
+  @Path("override-min-nodes")
   public Response setExtraLoad(
       @QueryParam("projectId") @Size(min = 1) final String projectId,
       @QueryParam("instanceId") @Size(min = 1) final String instanceId,
       @QueryParam("clusterId") @Size(min = 1) final String clusterId,
-      @QueryParam("loadDelta") final Integer loadDelta) {
+      @QueryParam("minNodesOverride") final Integer minNodesOverride) {
+
     final BigtableCluster cluster =
         new BigtableClusterBuilder()
             .projectId(projectId)
             .instanceId(instanceId)
             .clusterId(clusterId)
-            .minNodes(0)
-            .maxNodes(0)
-            .cpuTarget(0)
-            .overloadStep(Optional.of(0))
-            .enabled(true)
-            .loadDelta(loadDelta)
+            .minNodesOverride(minNodesOverride)
             .build();
     try {
       BigtableUtil.pushContext(cluster);
-      if (database.updateLoadDelta(projectId, instanceId, clusterId, loadDelta)) {
-        LOGGER.info("cluster loadDelta updated to {}", loadDelta);
+      final Optional<BigtableCluster> maybeCluster =
+          database.getBigtableCluster(projectId, instanceId, clusterId);
+      if (!maybeCluster.isPresent()) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      if (minNodesOverride > maybeCluster.get().maxNodes()) {
```

Comment:
I don't think this should be an error. Caller does not and should not care about the max nodes configured in the Autoscaler. It's autoscalers responsibility to not scale up beyond its max nodes. So it should accept the large minNodesOverride, and make sure to not scale up in `AutoscaleJob.sizeConstraints` where it checks precisely for this. (Also someone can go and manually increase the max nodes at a later time)

Reply:
Scenario:
user requests 20 extra nodes, but we can add only 3 since we've maxed out. 
so, we have 2 options:
1. silently run at max nodes capacity and in all probability bigtable can't handle the load and the job fails after X minutes.
2. fail explicitly (hopefully at jobAnnounce itself) and we say something like "autoscaler can't scale that much - so fix your max_nodes or requested_nodes and retry"

I prefer the second approach, its the hygienic thing to do.

This is different from the normal autoscaler hitting size constraints. I definitely do not want my bigtable to grind to a halt after X minutes of running a dataflow job.

(This also gives me the thought that we are not that explicit with the response message. We should be adding a proper status message rather than just response code.)

Reply:
@balaji for now I've reverted to what @gizem969 is saying, let's chat about this.